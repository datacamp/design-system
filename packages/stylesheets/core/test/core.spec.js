const dirTree = require('directory-tree');
const fs = require('fs');
const omitDeep = require('omit-deep-lodash');
const _ = require('lodash');

// remove hash from filename so snapshots are diffed rather than recreated
const replaceFileName = fileName =>
  fileName.replace(/(waffles)\.(\w+)(\.css)/, '$1$3');

describe('Core', () => {
  it('has correct file structure', () => {
    const dir = dirTree('lib');
    const structure = omitDeep(dir, 'size');
    const cssDirIndex = _.findIndex(
      structure.children,
      ({ name }) => name === 'css'
    );
    const cssFiles = structure.children[cssDirIndex].children;
    structure.children[cssDirIndex].children = cssFiles.map(file => ({
      ...file,
      name: replaceFileName(file.name),
      path: replaceFileName(file.path),
    }));
    expect(structure).toMatchSnapshot();
  });

  it('generates the correct files', () => {
    dirTree('lib', { extensions: /\.(css|scss|json)$/ }, item => {
      const file =
        item.extension === '.json'
          ? require(`../${item.path}`) // eslint-disable-line
          : replaceFileName(fs.readFileSync(item.path, 'utf8'));
      expect(file).toMatchSnapshot(replaceFileName(item.path));
    });
  });
});
