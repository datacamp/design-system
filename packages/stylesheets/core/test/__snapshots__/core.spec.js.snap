// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core has correct file structure 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "extension": ".css",
          "name": "core.waffles.css",
          "path": "lib/css/core.waffles.css",
          "type": "file",
        },
        Object {
          "extension": ".css",
          "name": "index.waffles.css",
          "path": "lib/css/index.waffles.css",
          "type": "file",
        },
      ],
      "name": "css",
      "path": "lib/css",
      "type": "directory",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "extension": ".svg",
              "name": "add.svg",
              "path": "lib/icons/svgs/add.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "apple.svg",
              "path": "lib/icons/svgs/apple.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "archive.svg",
              "path": "lib/icons/svgs/archive.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_2_down.svg",
              "path": "lib/icons/svgs/arrow_2_down.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_2_left.svg",
              "path": "lib/icons/svgs/arrow_2_left.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_2_right.svg",
              "path": "lib/icons/svgs/arrow_2_right.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_2_up.svg",
              "path": "lib/icons/svgs/arrow_2_up.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_down.svg",
              "path": "lib/icons/svgs/arrow_down.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_left.svg",
              "path": "lib/icons/svgs/arrow_left.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_left_double.svg",
              "path": "lib/icons/svgs/arrow_left_double.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_right.svg",
              "path": "lib/icons/svgs/arrow_right.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_right_double.svg",
              "path": "lib/icons/svgs/arrow_right_double.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "arrow_up.svg",
              "path": "lib/icons/svgs/arrow_up.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "assignment.svg",
              "path": "lib/icons/svgs/assignment.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "attachment.svg",
              "path": "lib/icons/svgs/attachment.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "attention.svg",
              "path": "lib/icons/svgs/attention.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "audio.svg",
              "path": "lib/icons/svgs/audio.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "bars.svg",
              "path": "lib/icons/svgs/bars.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "book.svg",
              "path": "lib/icons/svgs/book.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "building.svg",
              "path": "lib/icons/svgs/building.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "bullet.svg",
              "path": "lib/icons/svgs/bullet.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "calendar.svg",
              "path": "lib/icons/svgs/calendar.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "chat.svg",
              "path": "lib/icons/svgs/chat.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "checkmark.svg",
              "path": "lib/icons/svgs/checkmark.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "checkmark_circle.svg",
              "path": "lib/icons/svgs/checkmark_circle.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "circle-stroke.svg",
              "path": "lib/icons/svgs/circle-stroke.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "circle.svg",
              "path": "lib/icons/svgs/circle.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "clock.svg",
              "path": "lib/icons/svgs/clock.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "close.svg",
              "path": "lib/icons/svgs/close.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "code-brackets.svg",
              "path": "lib/icons/svgs/code-brackets.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "command.svg",
              "path": "lib/icons/svgs/command.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "company.svg",
              "path": "lib/icons/svgs/company.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "compass.svg",
              "path": "lib/icons/svgs/compass.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "copy.svg",
              "path": "lib/icons/svgs/copy.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "course.svg",
              "path": "lib/icons/svgs/course.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "credit_card.svg",
              "path": "lib/icons/svgs/credit_card.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "currency.svg",
              "path": "lib/icons/svgs/currency.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "custom_track.svg",
              "path": "lib/icons/svgs/custom_track.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dashboard.svg",
              "path": "lib/icons/svgs/dashboard.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "database.svg",
              "path": "lib/icons/svgs/database.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "datacamp.svg",
              "path": "lib/icons/svgs/datacamp.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dc-icon-academic.svg",
              "path": "lib/icons/svgs/dc-icon-academic.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dc-icon-datacamp-filled.svg",
              "path": "lib/icons/svgs/dc-icon-datacamp-filled.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dc-icon-enterprise.svg",
              "path": "lib/icons/svgs/dc-icon-enterprise.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dc-icon-mobile-phone.svg",
              "path": "lib/icons/svgs/dc-icon-mobile-phone.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dc-icon-prerequisite.svg",
              "path": "lib/icons/svgs/dc-icon-prerequisite.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "dc-icon-professional.svg",
              "path": "lib/icons/svgs/dc-icon-professional.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "download.svg",
              "path": "lib/icons/svgs/download.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "education.svg",
              "path": "lib/icons/svgs/education.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "error.svg",
              "path": "lib/icons/svgs/error.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "exercise.svg",
              "path": "lib/icons/svgs/exercise.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "expand.svg",
              "path": "lib/icons/svgs/expand.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "eye.svg",
              "path": "lib/icons/svgs/eye.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "facebook.svg",
              "path": "lib/icons/svgs/facebook.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "facebook_square.svg",
              "path": "lib/icons/svgs/facebook_square.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "filter.svg",
              "path": "lib/icons/svgs/filter.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "flagpole.svg",
              "path": "lib/icons/svgs/flagpole.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "github.svg",
              "path": "lib/icons/svgs/github.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "globe.svg",
              "path": "lib/icons/svgs/globe.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "google_play.svg",
              "path": "lib/icons/svgs/google_play.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "google_plus.svg",
              "path": "lib/icons/svgs/google_plus.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "google_plus_square.svg",
              "path": "lib/icons/svgs/google_plus_square.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "group.svg",
              "path": "lib/icons/svgs/group.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "hint.svg",
              "path": "lib/icons/svgs/hint.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "home.svg",
              "path": "lib/icons/svgs/home.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "incorrect.svg",
              "path": "lib/icons/svgs/incorrect.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "info.svg",
              "path": "lib/icons/svgs/info.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "info_2.svg",
              "path": "lib/icons/svgs/info_2.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "lab.svg",
              "path": "lib/icons/svgs/lab.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "learn.svg",
              "path": "lib/icons/svgs/learn.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "library.svg",
              "path": "lib/icons/svgs/library.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "lightbulb.svg",
              "path": "lib/icons/svgs/lightbulb.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "link.svg",
              "path": "lib/icons/svgs/link.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "linkedin.svg",
              "path": "lib/icons/svgs/linkedin.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "linkedin_square.svg",
              "path": "lib/icons/svgs/linkedin_square.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "location.svg",
              "path": "lib/icons/svgs/location.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "lock.svg",
              "path": "lib/icons/svgs/lock.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "mail.svg",
              "path": "lib/icons/svgs/mail.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "markdown.svg",
              "path": "lib/icons/svgs/markdown.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "math.svg",
              "path": "lib/icons/svgs/math.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "micro.svg",
              "path": "lib/icons/svgs/micro.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "more.svg",
              "path": "lib/icons/svgs/more.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "multiple-choice.svg",
              "path": "lib/icons/svgs/multiple-choice.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "news.svg",
              "path": "lib/icons/svgs/news.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "notification.svg",
              "path": "lib/icons/svgs/notification.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "paypal_logo.svg",
              "path": "lib/icons/svgs/paypal_logo.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "pdf.svg",
              "path": "lib/icons/svgs/pdf.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "phone.svg",
              "path": "lib/icons/svgs/phone.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "play.svg",
              "path": "lib/icons/svgs/play.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "project.svg",
              "path": "lib/icons/svgs/project.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "python.svg",
              "path": "lib/icons/svgs/python.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "r.svg",
              "path": "lib/icons/svgs/r.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "redo.svg",
              "path": "lib/icons/svgs/redo.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "reply.svg",
              "path": "lib/icons/svgs/reply.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "reporting.svg",
              "path": "lib/icons/svgs/reporting.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "rss.svg",
              "path": "lib/icons/svgs/rss.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "search.svg",
              "path": "lib/icons/svgs/search.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "settings.svg",
              "path": "lib/icons/svgs/settings.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slack.svg",
              "path": "lib/icons/svgs/slack.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-final.svg",
              "path": "lib/icons/svgs/slide-final.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-full.svg",
              "path": "lib/icons/svgs/slide-full.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-title.svg",
              "path": "lib/icons/svgs/slide-title.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-two-col.svg",
              "path": "lib/icons/svgs/slide-two-col.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-two-row-two-col.svg",
              "path": "lib/icons/svgs/slide-two-row-two-col.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-two-rows-wide.svg",
              "path": "lib/icons/svgs/slide-two-rows-wide.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "slide-two-rows.svg",
              "path": "lib/icons/svgs/slide-two-rows.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "sparkles.svg",
              "path": "lib/icons/svgs/sparkles.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "spinner.svg",
              "path": "lib/icons/svgs/spinner.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "star.svg",
              "path": "lib/icons/svgs/star.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "stats.svg",
              "path": "lib/icons/svgs/stats.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "stopwatch.svg",
              "path": "lib/icons/svgs/stopwatch.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "tab.svg",
              "path": "lib/icons/svgs/tab.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "tag.svg",
              "path": "lib/icons/svgs/tag.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "team.svg",
              "path": "lib/icons/svgs/team.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "theme-toggle.svg",
              "path": "lib/icons/svgs/theme-toggle.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "trash.svg",
              "path": "lib/icons/svgs/trash.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "trim-begin.svg",
              "path": "lib/icons/svgs/trim-begin.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "trophy.svg",
              "path": "lib/icons/svgs/trophy.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "twitter.svg",
              "path": "lib/icons/svgs/twitter.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "upload.svg",
              "path": "lib/icons/svgs/upload.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "user.svg",
              "path": "lib/icons/svgs/user.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "user_2.svg",
              "path": "lib/icons/svgs/user_2.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-forward.svg",
              "path": "lib/icons/svgs/video-forward.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-next-slide.svg",
              "path": "lib/icons/svgs/video-next-slide.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-pause.svg",
              "path": "lib/icons/svgs/video-pause.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-play.svg",
              "path": "lib/icons/svgs/video-play.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-prev-slide.svg",
              "path": "lib/icons/svgs/video-prev-slide.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-reverse.svg",
              "path": "lib/icons/svgs/video-reverse.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-slides.svg",
              "path": "lib/icons/svgs/video-slides.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-trim-begin.svg",
              "path": "lib/icons/svgs/video-trim-begin.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video-trim-end.svg",
              "path": "lib/icons/svgs/video-trim-end.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "video.svg",
              "path": "lib/icons/svgs/video.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "write.svg",
              "path": "lib/icons/svgs/write.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "xp.svg",
              "path": "lib/icons/svgs/xp.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "youtube.svg",
              "path": "lib/icons/svgs/youtube.svg",
              "type": "file",
            },
            Object {
              "extension": ".svg",
              "name": "youtube_logo.svg",
              "path": "lib/icons/svgs/youtube_logo.svg",
              "type": "file",
            },
          ],
          "name": "svgs",
          "path": "lib/icons/svgs",
          "type": "directory",
        },
        Object {
          "extension": ".svg",
          "name": "symbols.svg",
          "path": "lib/icons/symbols.svg",
          "type": "file",
        },
      ],
      "name": "icons",
      "path": "lib/icons",
      "type": "directory",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".js",
          "name": "iconNames.js",
          "path": "lib/js/iconNames.js",
          "type": "file",
        },
        Object {
          "extension": ".js",
          "name": "index.js",
          "path": "lib/js/index.js",
          "type": "file",
        },
      ],
      "name": "js",
      "path": "lib/js",
      "type": "directory",
    },
    Object {
      "children": Array [
        Object {
          "extension": ".scss",
          "name": "core-ingredients.scss",
          "path": "lib/scss/core-ingredients.scss",
          "type": "file",
        },
        Object {
          "children": Array [
            Object {
              "extension": ".scss",
              "name": "_base.scss",
              "path": "lib/scss/design/_base.scss",
              "type": "file",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_animations.scss",
                  "path": "lib/scss/design/animations/_animations.scss",
                  "type": "file",
                },
              ],
              "name": "animations",
              "path": "lib/scss/design/animations",
              "type": "directory",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_forms-legacy.scss",
                  "path": "lib/scss/design/forms/_forms-legacy.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_forms.scss",
                  "path": "lib/scss/design/forms/_forms.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_input-checkbox.scss",
                  "path": "lib/scss/design/forms/_input-checkbox.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_input-group.scss",
                  "path": "lib/scss/design/forms/_input-group.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_input-radio.scss",
                  "path": "lib/scss/design/forms/_input-radio.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_input-search.scss",
                  "path": "lib/scss/design/forms/_input-search.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_input-toggle-switch.scss",
                  "path": "lib/scss/design/forms/_input-toggle-switch.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_select.scss",
                  "path": "lib/scss/design/forms/_select.scss",
                  "type": "file",
                },
              ],
              "name": "forms",
              "path": "lib/scss/design/forms",
              "type": "directory",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_grid-flex.scss",
                  "path": "lib/scss/design/layout/_grid-flex.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_grid.scss",
                  "path": "lib/scss/design/layout/_grid.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_layout.scss",
                  "path": "lib/scss/design/layout/_layout.scss",
                  "type": "file",
                },
              ],
              "name": "layout",
              "path": "lib/scss/design/layout",
              "type": "directory",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_buttons.scss",
                  "path": "lib/scss/design/legacy/_buttons.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_globals.scss",
                  "path": "lib/scss/design/legacy/_globals.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_hr-line.scss",
                  "path": "lib/scss/design/legacy/_hr-line.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_kill.scss",
                  "path": "lib/scss/design/legacy/_kill.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_legacy.scss",
                  "path": "lib/scss/design/legacy/_legacy.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_shim.scss",
                  "path": "lib/scss/design/legacy/_shim.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_tables.scss",
                  "path": "lib/scss/design/legacy/_tables.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_tooltip.scss",
                  "path": "lib/scss/design/legacy/_tooltip.scss",
                  "type": "file",
                },
              ],
              "name": "legacy",
              "path": "lib/scss/design/legacy",
              "type": "directory",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_afh.scss",
                  "path": "lib/scss/design/mixins/_afh.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_blocks.scss",
                  "path": "lib/scss/design/mixins/_blocks.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_buttons.scss",
                  "path": "lib/scss/design/mixins/_buttons.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_links.scss",
                  "path": "lib/scss/design/mixins/_links.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_lists.scss",
                  "path": "lib/scss/design/mixins/_lists.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_mixins.scss",
                  "path": "lib/scss/design/mixins/_mixins.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_responsive.scss",
                  "path": "lib/scss/design/mixins/_responsive.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_state.scss",
                  "path": "lib/scss/design/mixins/_state.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_stretch.scss",
                  "path": "lib/scss/design/mixins/_stretch.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_text.scss",
                  "path": "lib/scss/design/mixins/_text.scss",
                  "type": "file",
                },
              ],
              "name": "mixins",
              "path": "lib/scss/design/mixins",
              "type": "directory",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_link.scss",
                  "path": "lib/scss/design/typography/_link.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_typography-legacy.scss",
                  "path": "lib/scss/design/typography/_typography-legacy.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_typography.scss",
                  "path": "lib/scss/design/typography/_typography.scss",
                  "type": "file",
                },
              ],
              "name": "typography",
              "path": "lib/scss/design/typography",
              "type": "directory",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_utilities.scss",
                  "path": "lib/scss/design/utilities/_utilities.scss",
                  "type": "file",
                },
              ],
              "name": "utilities",
              "path": "lib/scss/design/utilities",
              "type": "directory",
            },
            Object {
              "extension": ".scss",
              "name": "variables.scss",
              "path": "lib/scss/design/variables.scss",
              "type": "file",
            },
            Object {
              "children": Array [
                Object {
                  "extension": ".scss",
                  "name": "_feixen-font.scss",
                  "path": "lib/scss/design/vendor/_feixen-font.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_jetbrains-mono-font.scss",
                  "path": "lib/scss/design/vendor/_jetbrains-mono-font.scss",
                  "type": "file",
                },
                Object {
                  "extension": ".scss",
                  "name": "_normalize.scss",
                  "path": "lib/scss/design/vendor/_normalize.scss",
                  "type": "file",
                },
              ],
              "name": "vendor",
              "path": "lib/scss/design/vendor",
              "type": "directory",
            },
          ],
          "name": "design",
          "path": "lib/scss/design",
          "type": "directory",
        },
        Object {
          "extension": ".scss",
          "name": "fonts.scss",
          "path": "lib/scss/fonts.scss",
          "type": "file",
        },
        Object {
          "extension": ".scss",
          "name": "ingredients.scss",
          "path": "lib/scss/ingredients.scss",
          "type": "file",
        },
        Object {
          "extension": ".scss",
          "name": "normalize.scss",
          "path": "lib/scss/normalize.scss",
          "type": "file",
        },
      ],
      "name": "scss",
      "path": "lib/scss",
      "type": "directory",
    },
    Object {
      "extension": ".json",
      "name": "tokens.json",
      "path": "lib/tokens.json",
      "type": "file",
    },
  ],
  "name": "lib",
  "path": "lib",
  "type": "directory",
}
`;
